R/codecov.R

```diff
@@ -0,0 +1,190 @@
+                         "&build=", Sys.getenv('DRONE_BUILD_NUMBER'),
+                         "&build_url=", Sys.getenv('DRONE_BUILD_URL'),
+                         "&commit=", Sys.getenv('DRONE_COMMIT')), sep="")
+  # --------
+  # AppVeyor
+  # --------
+  } else if (Sys.getenv('CI') == "True" && Sys.getenv('APPVEYOR') == 'True') {
+    # http://www.appveyor.com/docs/environment-variables
+    codecov_url <- paste(base_url, "/upload/v2?service=AppVeyor",
+                         "&branch=", Sys.getenv('APPVEYOR_REPO_BRANCH'),
+                         "&build=", Sys.getenv('APPVEYOR_BUILD_NUMBER'),
+                         "&owner=", strsplit(Sys.getenv('APPVEYOR_REPO_NAME'), '/')[0],
+                         "&repo=", strsplit(Sys.getenv('APPVEYOR_REPO_NAME'), '/')[1],
+                         "&commit=", Sys.getenv('APPVEYOR_REPO_COMMIT')), sep="")
+  # -------
+  # Wercker
+  # -------
+  } else if (Sys.getenv('CI') == "true" && Sys.getenv('WERCKER_GIT_BRANCH')) {
+    # http://devcenter.wercker.com/articles/steps/variables.html
+    codecov_url <- paste(base_url, "/upload/v2?service=werker",
+                         "&branch=", Sys.getenv('WERCKER_GIT_BRANCH'),
+                         "&build=", Sys.getenv('WERCKER_MAIN_PIPELINE_STARTED'),
+                         "&owner=", Sys.getenv('WERCKER_GIT_OWNER'),
+                         "&repo=", Sys.getenv('WERCKER_GIT_REPOSITORY'),
+                         "&commit=", Sys.getenv('WERCKER_GIT_COMMIT')), sep="")
+
+  # ---------
+  # Local GIT
+  # ---------
+  } else {
+    branch <- gsub("^\\s+|\\s+$", "", system("git rev-parse --abbrev-ref HEAD", intern = TRUE))
+    codecov_url <- paste(base_url, "/upload/v2",
+                         "&branch=", ifelse(branch=='HEAD', 'master', branch),
+                         "&commit=", gsub("^\\s+|\\s+$", "", system('git rev-parse HEAD', intern = TRUE)), sep="")
+  }
+
+  if (Sys.getenv('CODECOV_TOKEN')) {
+    codecov_url <- paste(codecov_url, "&token=", Sys.getenv('CODECOV_TOKEN'), sep="")
+  }
+
+  # add query args
+  # Sys.getenv("TRAVIS_JOB_ID"), service = "travis"
+  coverage <- to_codecov(package_coverage(path, relative_path = TRUE, ...))
+
+  name <- tempfile()
+  con <- file(name)
+  writeChar(con = con, coverage, eos = NULL)
+  close(con)
+  on.exit(unlink(name))
+  httr::content(httr::POST(codecov_url, body = list(json_file = httr::upload_file(name))))
+}
+
+to_codecov <- function(x) {
+  coverages <- per_line(x)
+
+  coverage_names <- names(coverages)
+
+  if (!is.null(attr(x, "path"))) {
+    coverage_names <- file.path(attr(x, "path"), coverage_names)
+  }
+
+  res <- mapply(
+    function(name, source, coverage) {
+      list("name" = jsonlite::unbox(name),
+        "coverage" = coverage)
+    },
+    coverage_names,
+    sources,
+    coverages,
+    SIMPLIFY = FALSE,
+    USE.NAMES = FALSE)
+
+  jsonlite::toJSON(na = "null", list("files" = res, "uploader" = "R"))
+}
+
+per_line <- function(x) {
```

Comment:
You do not have to copy this function from the `coveralls.R` file, this function shares the same namespace.